python/ml4ir/base/features/preprocessing.py

```diff
@@ -290,6 +291,34 @@ def convert_label_to_clicks(label_vector, dtype):
     clicks = tf.dtypes.cast(cond, typ)
     return clicks
 
+@tf.function
+def convert_fr_to_one_hot(fr_value, max_ranks, mask_fr):
```

Comment:
@mohazahran I don't think you should do this as a preprocessing step. Let's discuss offline if you have questions.